config/kernel.m4

```diff
@@ -681,11 +681,16 @@ AC_DEFUN([ZFS_LINUX_COMPILE], [
 		building kernel modules])
 	AC_ARG_VAR([KERNEL_LLVM], [Binary option to
 		build kernel modules with LLVM/CLANG toolchain])
+	AC_ARG_VAR([KERNEL_CROSS_COMPILE], [Cross compile prefix
+		for kernel module builds])
+	AC_ARG_VAR([KERNEL_ARCH], [Architecture to build kernel modules for])
 	AC_TRY_COMMAND([
 	    KBUILD_MODPOST_NOFINAL="$5" KBUILD_MODPOST_WARN="$6"
 	    make modules -k -j$TEST_JOBS ${KERNEL_CC:+CC=$KERNEL_CC}
 	    ${KERNEL_LD:+LD=$KERNEL_LD} ${KERNEL_LLVM:+LLVM=$KERNEL_LLVM}
 	    CONFIG_MODULES=y CFLAGS_MODULE=-DCONFIG_MODULES
+	    ${KERNEL_CROSS_COMPILE:+CROSS_COMPILE=$KERNEL_CROSS_COMPILE}
+	    ${KERNEL_ARCH:+ARCH=$KERNEL_ARCH}
```

Comment:
To support cross-compiling RPM packages you'll need to also add `KERNEL_CROSS_COMPILE` and `KERNEL_ARCH` to the `ZFS_AC_RPM` macro in `config/zfs-build.m4`.  We already do the same thing for the other `KERNEL_` variables.

Reply:
Thanks for the pointer, Brian! I added the extra bits to zfs-build.m4 and the two kmod spec files. Could you please review? Note that I did not test these changes at all.

Reply:
Thanks for the quick turn around, that looks good.